src/pybel/io/umbrella_nodelink_json.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+"""
+Conversion functions for customized node-link JSON format prior canonicalization. As an alternative to previous JSON
+export functions, PyBEL also provides a customized JSON exporter prior node canonicalization. This module uses the
+'pybel.canonicalize' module prior exporting to the node-link JSON format. This allows the inclusion of modifiers in the
+nodes instead of the edges, the default schema in PyBEL.
+"""
+
+import json
+from itertools import chain, count
+from typing import Any, Mapping, TextIO, Union
+
+from networkx.utils import open_file
+
+from ..canonicalize import _decanonicalize_edge_node, edge_to_tuple
+from ..constants import GRAPH_ANNOTATION_LIST, OBJECT, SUBJECT
+from ..struct import BELGraph
+
+__all__ = [
+    'umbrella_nodelink_json',
+    'to_umbrella_nodelink_json_file',
+]
+
+
+def to_umbrella_nodelink_json(graph: BELGraph) -> Mapping[str, Any]:
+    """Convert this graph to a node-link JSON object by previously canonicalizing the nodes.
```

Comment:
Update docstring to be more descriptive of what this actually is and *why* you would want to do this